src/agent/main.cpp

```diff
@@ -248,7 +261,16 @@ int main(int argc, char *argv[])
 #if OPENTHREAD_ENABLE_COVERAGE
         __gcov_flush();
 #endif
-        execvp(argv[0], argv);
+        std::vector<std::string> args(argv, argv + argc);
```

Comment:
```suggestion
        std::vector<const char *> args(argv, argv + argc);
```

nit: seems simpler to use `const char *` directly.

Reply:
Done.